### PR TITLE
Remove two unused methods in Iterables

### PR DESCRIPTION
**Pull request by cbuescher**
These helper methods are unused in the rest of the codebase.

### CODE REVIEW

**Review by original-brownbear — APPROVED** [review]
LGTM assuming CI goes green, one small nit.

File: modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/CustomReflectionObjectHandler.java [hunk 1]
```diff
@@ -20,16 +20,17 @@
 package org.elasticsearch.script.mustache;
 
 import com.github.mustachejava.reflect.ReflectionObjectHandler;
+
```
[the remark this answers is not in the document]
Comment (cbuescher) [reply modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/CustomReflectionObjectHandler.java]: sure

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by Conky5** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@elasticmachine run the gradle build tests 1

**Comment by cbuescher** [comment]
@elasticmachine run the gradle build tests 2

**Comment by cbuescher** [comment]
@elasticmachine run the gradle build tests 2

**Pull request merged**